src/components/container/SwellingCalendar/SwellingCalendar.tsx

```diff
@@ -0,0 +1,177 @@
+                background: 'var(--mdhui-color-danger)'
+            },
+            streak: true,
+            streakColor: '#FCCCC7'
+        },
+    };
+
+    const computeStateForDay = (date: Date): string => {
+        let key : string = startOfDay(date).toDateString();
+        return (data && data?.get(key)?.severity) ?? "";
+    };
+
+    function transformToCalendarData(results : SurveyAnswer[]){
+        var swellingEntries : Map<Guid, SwellingDay> = createResultsMap(results);
+
+        var calendarData : Map<string, SwellingDay> = new Map<string, SwellingDay>();
+        swellingEntries.forEach( swellingEntry => {
+            let key : string = startOfDay(parseISO(swellingEntry.dateRecorded)).toDateString();
+            let exists = calendarData.get(key);
+            if (exists){
+                if (parseISO(swellingEntry.dateRecorded) > parseISO(exists.dateRecorded)){
+                    exists.severity = swellingEntry.severity;
+                    calendarData.set(key, exists);
+                }
+            }
+            else {
+                calendarData.set(key, swellingEntry);
+            }
+        });
+
+        setData(calendarData);
+    }
+
+    function createResultsMap(results : SurveyAnswer[]){
+        var groupedByResult = new  Map<Guid, SwellingDay>();
+
+        results.forEach( (r) => {
+            var exists = groupedByResult.get(r.surveyResultID);
+            var swellingDay = exists ?? {dateRecorded : "", severity : ""};
+
+            if (r.resultIdentifier === props.dateRecordedResultIdentifier && r.answers)
+            {
+                swellingDay.dateRecorded = r.answers[0];
+            }
+
+            if (r.resultIdentifier === props.severityResultIdentifier && r.answers)
+            {
+                var severity = "";
+                switch (r.answers[0]){
+                    case "mild":
+                    case "severity1":
+                        severity = "mild";
+                        break;
+                    case "moderate":
+                    case "severity2":
+                        severity = "moderate";
+                        break;
+                    case "severe":
+                    case "severity3":
+                        severity = "severe";
+                        break;
+                    }
+
+                    swellingDay.severity = severity;
+            }
+
+            groupedByResult.set(r.surveyResultID, swellingDay);
+        });
+
+        return groupedByResult;
+    }
+
+    const renderDay = (year: number, month: number, day?: number): React.JSX.Element => {
+        return <CalendarDay
+            year={year}
+            month={month}
+            day={day}
+            stateConfiguration={stateConfiguration}
+            computeStateForDay={computeStateForDay}
+        />;
+    };
+
+    function pageMonth(newStart: Date) {
+        setMonth(startOfMonth(newStart));
+    } 
+
+
+    useEffect(() => {
+        initialize();
```

Comment:
There is a helper hook you can use to get the same functionality as this `useEffect` setup.

```
useInitializeView(initialize);
```

However, I'm also not sure this is going to do anything meaningful for you.  I may be misunderstanding the logic, but I noticed that the `initialize` function only loads data if `data` has not been set yet.  It looks like `data` will be set in all cases after the first execution of `initialize()`.  Or am I missing something there?